Source/WebCore/PAL/pal/spi/mac/NSTextInputContextSPI.h

```diff
@@ -29,6 +29,24 @@
 
 #import <AppKit/NSTextInputContext_Private.h>
 
+// Staging for rdar://126696059
+@interface NSTextSelectionRect : NSObject
+@property (nonatomic, readonly) NSRect rect;
+@property (nonatomic, readonly) NSWritingDirection writingDirection;
+@property (nonatomic, readonly) BOOL isVertical;
+@property (nonatomic, readonly) NSAffineTransform *transform;
+@end
+
+@interface NSTextPlaceholder : NSObject
+@property (nonatomic, readonly) NSArray<NSTextSelectionRect *> *rects;
+@end
```

Comment:
I think `NSTextPlaceholder` is already in the SDK, so you could just import the header, and then declare this in a category.

Reply:
While this is true, if it's not going to hurt anything, I'd rather leave it for now, since this is all staging and will be gone soon anyways.

Reply:
I would've thought defining the interface again would result in a duplicate interface declaration 🤔.